### PR TITLE
[ICP-10363] Fibaro Temperature Sensor (for FGT) doesn't go offline

### PR DESCRIPTION
**Pull request by PKacprowiczS**
@greens @dkirker @KKlimczukS @MGoralczykS @ZWozniakS

### CODE REVIEW

**Comment by greens** [comment]
I think it might be worth adding that configurationGet to the `refreshChild` method as well.

File: devicetypes/smartthings/fibaro-heat-controller.src/fibaro-heat-controller.groovy [hunk 1]
```diff
@@ -328,4 +334,9 @@ private getMaxHeatingSetpointTemperature() {
 
 private getMinHeatingSetpointTemperature() {
 	temperatureScale == 'C' ? 10 : 50
+}
+
+private changeTemperatureSensorStatus(status) {
+	def map = [name: "healthStatus", value: status]
```
Comment (PKacprowiczS) [inline devicetypes/smartthings/fibaro-heat-controller.src/fibaro-heat-controller.groovy]: @greens this event doesn't mark device offline. What should I send instead?

Comment (greens) [reply devicetypes/smartthings/fibaro-heat-controller.src/fibaro-heat-controller.groovy]: `name: "DeviceWatch-DeviceStatus", value: "offline"`

I think that should work. I don't know how it meshes with devices that are actively pinged, but give it a shot.

File: devicetypes/smartthings/fibaro-heat-controller.src/fibaro-heat-controller.groovy [hunk 2]
```diff
@@ -299,7 +306,8 @@ def switchMode() {
 def sendEventToChild(event) {
 	String childDni = "${device.deviceNetworkId}:2"
 	def child = childDevices.find { it.deviceNetworkId == childDni }
-	child?.sendEvent(event)
+	if (state.isChildOnline)
+		child?.sendEvent(event)
```
Comment (greens) [inline devicetypes/smartthings/fibaro-heat-controller.src/fibaro-heat-controller.groovy:310]: were you having a problem with this being sent too often?

**Comment by dkirker** [comment]
It is looking good! One question: are we relying on ping() to be called for us to check on the child sensor?

File: devicetypes/smartthings/fibaro-heat-controller.src/fibaro-heat-controller.groovy (hunk 2 above)
Comment (PKacprowiczS) [reply devicetypes/smartthings/fibaro-heat-controller.src/fibaro-heat-controller.groovy:310]: Kind of, but it's more like, after I've received Multilevel Report from endpoint 2, I was sending an event to child, and it was bringing it back online every time, meaning that sending "offline" event haven't done a trick. So this flag is a workaround to store real child status in parent device and prevent sending 'fake' events to additional temperature sensor.

**Comment by PKacprowiczS** [comment]
@dkirker yep

**Comment by greens** [comment]
@PKacprowiczS want to change it back so we can merge?

**Comment by PKacprowiczS** [comment]
@greens I wanted to wait for all devices' issues to be verified, but it seems those are more related to its design, not DTH. Now names are back to normal.

**Review by greens — APPROVED** [review]

**Pull request merged**